mrtt-ui/src/components/PreRestorationAssessmentForm.js

```diff
@@ -0,0 +1,520 @@
+  const speciesCompositionWatcher = watch('speciesComposition')
+  const [isSubmitting, setisSubmitting] = useState(false)
+  const [isError, setIsError] = useState(false)
+  const [isLoading, setIsLoading] = useState(false)
+  const [mangroveSpeciesForCountriesSelected, setMangroveSpeciesForCountriesSelected] = useState([])
+  const [mangroveSpeciesTypesChecked, setMangroveSpeciesTypesChecked] = useState([])
+  const [showAddTabularInputRow, setShowAddTabularInputRow] = useState(false)
+
+  const loadServerData = useCallback(
+    (serverResponse) => {
+      const defaultMeasurementsTaken = [
+        { measurementType: 'Tidal range', measurementValue: '' },
+        { measurementType: 'Elevation to sea level', measurementValue: '' },
+        { measurementType: 'Water salinity', measurementValue: '' },
+        { measurementType: 'Soil pore water salinity', measurementValue: '' },
+        { measurementType: 'Water PH', measurementValue: '' },
+        { measurementType: 'Soil pore water PH', measurementValue: '' },
+        { measurementType: 'Soil type', measurementValue: '' },
+        { measurementType: 'Soil organic matter', measurementValue: '' }
+      ]
+
+      const siteCountriesResponse = getSiteCountries(serverResponse)
+
+      if (siteCountriesResponse.length) {
+        const countriesList = siteCountriesResponse.map(
+          (countryItem) => countryItem.properties.country
+        )
+        const species = []
+        countriesList.forEach((countrySelected) => {
+          mangroveSpeciesPerCountryList.forEach((countryItem) => {
+            if (countryItem.country.name === countrySelected) {
+              species.push(...countryItem.species)
+            }
+          })
+        })
+        const uniqueSpecies = [...new Set(species)]
+        setMangroveSpeciesForCountriesSelected(uniqueSpecies)
+      }
+      setMangroveSpeciesTypesChecked(getMangroveSpecies(serverResponse))
+
+      const physicalMeasurementsTakenInitialVal = getPhysicalMeasurementsTaken(serverResponse)
+
+      if (physicalMeasurementsTakenInitialVal.length === 0) {
+        physicalMeasurementsTakenReplace(defaultMeasurementsTaken)
+      }
+    },
+    [physicalMeasurementsTakenReplace]
+  )
+
+  useInitializeQuestionMappedForm({
+    apiUrl: apiAnswersUrl,
+    questionMapping: questionMapping.preRestorationAssessment,
+    resetForm,
+    setIsLoading,
+    successCallback: loadServerData
+  })
+
+  const handleSubmit = async (formData) => {
+    setisSubmitting(true)
+    setIsError(false)
+
+    if (!formData) return
+
+    axios
+      .patch(apiAnswersUrl, mapDataForApi('preRestorationAssessment', formData))
+      .then(() => {
+        setisSubmitting(false)
+        toast.success(language.success.submit)
+      })
+      .catch(() => {
+        setIsError(true)
+        setisSubmitting(false)
+        toast.error(language.error.submit)
+      })
+  }
+
+  const handleMangroveSpeciesPresentOnChange = (event, specie) => {
```

Comment:
lol, had to do `specie` for another project too.

Reply:
hah yeah I had to google what the singular was